examples/chip-tool/commands/pairing/OpenCommissioningWindowCommand.cpp

```diff
@@ -34,10 +34,13 @@ CHIP_ERROR OpenCommissioningWindowCommand::RunCommand()
     if (mCommissioningWindowOption == Controller::CommissioningWindowOpener::CommissioningWindowOption::kTokenWithRandomPIN)
     {
         SetupPayload ignored;
-        return mWindowOpener->OpenCommissioningWindow(mNodeId, System::Clock::Seconds16(mCommissioningWindowTimeout), mIteration,
-                                                      mDiscriminator, NullOptional, NullOptional,
-                                                      &mOnOpenCommissioningWindowCallback, ignored,
-                                                      /* readVIDPIDAttributes */ true);
+        return mWindowOpener->OpenCommissioningWindow(Controller::CommissioningWindowPasscodeParams()
+                                                          .SetNodeId(mNodeId)
+                                                          .SetTimeout(mCommissioningWindowTimeout)
+                                                          .SetIteration(mIteration)
+                                                          .SetDiscriminator(mDiscriminator)
+                                                          .SetReadVIDPIDAttributes(true),
+                                                      &mOnOpenCommissioningWindowCallback, ignored);
```

Comment:
why are the callbacks and setuppayload outside the params structure?

Reply:
It is mostly to make the params more generic so places other than this API like the[ RPC functions](https://github.com/project-chip/connectedhomeip/blob/b154a1b651f627afd514d4a476201ae62d3bdf7f/examples/fabric-bridge-app/linux/include/RpcClient.h#L58) which are passing these around internally shouldn't need yet another struct or having to define an ignored out param reference. Let me know if you have strong opinions on making it all part of the same class.

Reply:
why dow e allow RPC to dictate src/controller/CommissioningWindowOpener.h ?

RPC is not the main API in our SDK and it is an optional app-specific extension. src/controller/CommissioningWindowOpener.h is a SDK API, I would prefer to make that one clean to use.

Reply:
Moved the callback to params class in https://github.com/project-chip/connectedhomeip/pull/34209/commits/8862cacfbc9519b71496b65ff923ddbd06cb31dc